model/patch_lifecycle.go

```diff
@@ -491,6 +491,105 @@ func AbortPatchesWithGithubPatchData(createdBefore time.Time, closed bool, newPa
 	return errors.Wrap(catcher.Resolve(), "error aborting patches")
 }
 
+func MakeCommitQueueDescription(patches []patch.ModulePatch, projectRef *ProjectRef, project *Project) string {
+	commitFmtString := "'%s' into '%s/%s:%s'"
+	description := []string{}
+	for _, p := range patches {
+		// skip empty patches
+		if len(p.PatchSet.CommitMessages) == 0 {
+			continue
+		}
+		owner := projectRef.Owner
+		repo := projectRef.Repo
+		branch := projectRef.Branch
+		if p.ModuleName != "" {
+			module, err := project.GetModuleByName(p.ModuleName)
+			if err != nil {
+				continue
+			}
+			owner, repo = module.GetRepoOwnerAndName()
+			branch = module.Branch
+		}
+
+		description = append(description, fmt.Sprintf(commitFmtString, strings.Join(p.PatchSet.CommitMessages, " <- "), owner, repo, branch))
+	}
+
+	if len(description) == 0 {
+		description = []string{"No Commits Added"}
+	}
+
+	return "Commit Queue Merge: " + strings.Join(description, " || ")
+}
+
+func MakeMergePatchFromExisting(existingPatchID string, u *user.DBUser) (*patch.Patch, error) {
+	existingPatch, err := patch.FindOne(patch.ById(patch.NewId(existingPatchID)))
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting existing patch '%s'", existingPatchID)
+	}
+	if existingPatch == nil {
+		return nil, errors.Errorf("patch '%s' doesn't exist", existingPatchID)
+	}
+
+	// verify the patch and its modules are in mbox format
+	if !existingPatch.CanEnqueueToCommitQueue() {
```

Comment:
Should we also verify that the user who's merging is the user who created the original patch? (Or maybe a project admin)

Reply:
Why do you suggest we restrict who can enqueue it? It seems reasonable to allow you to merge my patch.

Reply:
Can an Evergreen user who doesn't have Github merge permissions for a project still merge a patch someone else wrote? If Github level permissions are checked elsewhere then I guess it's fine.

Reply:
That's an interesting point.
I'm not sure the CLI commit queue ever checks if an Evergreen user has permissions to merge.

Reply:
Oh haha i wonder if we should look into that. Okay though might as well go ahead with it then.

Reply:
Thanks!

Actually, I was thinking it over and I think I agree with your perspective: if I were to enqueue your patch the resulting commit would have your name on it (because the commits that come out of the patch will be `git am`ed from the commit metadata that came from your machine). Since fixing it to use my name on the commit is not feasible, the next best thing would probably be to prevent me from enqueueing your patch.

Reply:
Okay yeah I agree with that.

Reply:
done